packages/core/src/hydration/annotate.ts

```diff
@@ -399,7 +399,11 @@ function serializeLView(lView: LView, context: HydrationContext): SerializedView
     const i18nData = trySerializeI18nBlock(lView, i, context);
     if (i18nData) {
       ngh[I18N_DATA] ??= {};
-      ngh[I18N_DATA][noOffsetIndex] = i18nData;
+      ngh[I18N_DATA][noOffsetIndex] = i18nData.caseQueue;
+
+      ngh[DISCONNECTED_NODES] = Array.from(
+        new Set([...(ngh[DISCONNECTED_NODES] ?? []), ...i18nData.disconnectedNodes]),
+      );
```

Comment:
nit: it'd be great to have a helper method here, something like:
```suggestion
      ngh[DISCONNECTED_NODES] = arrayUnion(ngh[DISCONNECTED_NODES], i18nData.disconnectedNodes)
```

Reply:
There is also the `appendDisconnectedNodeIndex` helper function in this file, but it takes a `TNode` as an argument (to compute an index). We can probably do some refactoring to have a unified way of registering disconnected node ids.